Framework/DataObjects/src/TimeSplitter.cpp

```diff
@@ -277,5 +290,122 @@ TimeROI TimeSplitter::getTimeROI(const int workspaceIndex) {
+ *
+ * Events with masked times are allocated to destination index -1.
+ * @param events : list of input events
+ * @param partials : resulting partial lists of events
+ * @param pulseTof : if True, split according to Pulse + TOF time, otherwise split by Pulse time
+ * @param tofCorrect : rescale and shift the TOF values (factor*TOF + shift)
+ * @param factor : rescale the TOF values by a dimensionless factor.
+ * @param shift : shift the TOF values after rescaling, in units of microseconds.
+ * @throws RunTimeError : the event list is of type Mantid::API::EventType::WEIGHTED_NOTIME
+ */
+void TimeSplitter::splitEventList(const EventList &events, std::map<int, EventList *> partials, bool pulseTof,
+                                  bool tofCorrect, double factor, double shift) const {
+
+  if (events.getEventType() == EventType::WEIGHTED_NOTIME)
+    throw std::runtime_error("EventList::splitByTime() called on an EventList "
+                             "that no longer has time information.");
+
+  // Initialize the detector ID's and event type of the destination event lists
+  events.initializePartials(partials);
+
+  if (this->empty())
+    return;
+
+  // sort the list in-place
+  if (pulseTof)
+    // this sorting is preserved under linear transformation tof --> factor*tof+shift with factor>0
+    events.sortPulseTimeTOF();
+  else
+    events.sortPulseTime();
+
+  // fetch the times associated to each event
+  std::vector<DateAndTime> times;
+  if (pulseTof)
+    if (tofCorrect)
+      times = events.getPulseTOFTimesAtSample(factor, shift);
+    else
+      times = events.getPulseTOFTimes();
+  else
+    times = events.getPulseTimes();
+
+  // split the events
+  switch (events.getEventType()) {
+  case EventType::TOF:
+    this->splitEventVec(times, events.getEvents(), partials);
+    break;
+  case EventType::WEIGHTED:
+    this->splitEventVec(times, events.getWeightedEvents(), partials);
+    break;
+  default:
+    throw std::runtime_error("Unhandled event type");
+  }
+}
+
+/**
+ * Distribute a list of events by comparing a vector of times against the splitter boundaries.
+ *
+ * Each event in `events` has a corresponding time in `times`, which we use to find a destination index
+ * in the TimeSplitter object. The destination index is the key to find the target event list
+ * in the partials map.
+ *
+ * @tparam EVENTTYPE : one of EventType::TOF or EventType::WEIGHTED
+ * @param times : times associated to the events, used to find the destination index
+ * @param events : list of input times
+ * @param partials : target list of partial event lists, associated to the different destination indexes
+ * @throws : if the size of times and events are different
+ */
+template <typename EVENTTYPE>
+void TimeSplitter::splitEventVec(const std::vector<DateAndTime> &times, const std::vector<EVENTTYPE> &events,
+                                 std::map<int, EventList *> partials) const {
+  assert(times.size() == events.size());
```

Comment:
This only gets compiled/run for debug builds. I don't know if you were relying on it for release builds.

Reply:
dang it. I will change it to an `if(...){ throw(...) }`